### ISSUE TITLE
Authentication with PHP

### ISSUE THREAD

**Issue by clickstudio** [ftexchange/ftx#12]
I can't get a successful authentication using PHP. All I get is the following:

`(
    [error] => Not logged in
    [success] => 
)`

My code below. What am I doing wrong here ?

`$end_point = 'https://ftx.com/api/' . $args['end_point'];
  $ts = intval(microtime(true) * 1000);
  $signature = hash_hmac('sha256', $ts . 'GET' . '/' . $args['end_point'], $api_secret);
  
  $curl = curl_init();
  $headers[] = 'Accept: application/json';
  $headers[] = 'FTX-KEY: ' . $api_key;
  $headers[] = 'FTX-TS: ' . $ts;
  $headers[] = 'FTX-SIGN: ' . $signature;
  curl_setopt_array($curl, [
    CURLOPT_RETURNTRANSFER => 1,
    CURLOPT_HTTPHEADER => $headers,
    CURLOPT_URL => $end_point,
  ]);

  $resp = curl_exec($curl);
  $response = json_decode($resp, true);

  return $response;`

**Comment by vdbelt** [comment]
You'll need to use the full path in the signature payload. Prepend it with /api and your example should be good.

FTX may want to update the text example in the [docs](https://docs.ftx.com/#authentication). I found it confusing as well.

**Comment by ryuchix** [comment]
@clickstudio did you solve this? im experiencing the same output.

**Comment by penuel-leo** [comment]
> You'll need to use the full path in the signature payload. Prepend it with /api and your example should be good.
> 
> FTX may want to update the text example in the [docs](https://docs.ftx.com/#authentication). I found it confusing as well.
> 
> Edit: Also: `time()*1000`

Yes, replace '/wallet/balances' to '/api/wallet/balances' , everything is ok.

**Comment by allartdo** [comment]
@penuel-leo  Any idea's on how to do this with an POST request? I have the GET, post just doesn't want to work... See my code here: https://stackoverflow.com/questions/68550710/how-to-do-a-php-post-request-for-ftx-rest-api
